avocado/plugins/resolvers.py

```diff
@@ -196,3 +198,102 @@ def resolve(self, reference):
             return criteria_check
 
         return self._validate_and_load_runnables(reference)
+
+
+class ExecRunnablesRecipeInit(Init):
+    name = "exec-runnables-recipe"
+    description = 'Configuration for resolver plugin "exec-runnables-recipe" plugin'
+
+    def initialize(self):
+        help_msg = (
+            'Whether resolvers (such as "exec-runnables-recipe") should '
+            "execute files given as test references that have executable "
+            "permissions. This is disabled by default due to security "
+            "implications of running executables that may not be trusted."
+        )
+        settings.register_option(
+            section="resolver",
+            key="run_executables",
+            key_type=bool,
+            default=False,
+            help_msg=help_msg,
+        )
+
+        help_msg = (
+            "Command line options (space separated) that will be added "
+            "to the executable when executing it as a producer of "
+            "runnables-recipe JSON content."
+        )
+        settings.register_option(
```

Comment:
I am not sure if this is the best solution, what would happen when there would be more than one `run-executables-resolver`, all of them should get the same arguments?

What about have the arguments as part of the reference behind `:`. Like this:
```
$ avocado list --resolver-run-executables examples/nrunner/resolvers/exec_runnables_recipe_kind.sh:tap
```

Reply:
> I am not sure if this is the best solution, what would happen when there would be more than one `run-executables-resolver`, all of them should get the same arguments?
> 

Yes, the use case right now requires just a single (the same) argument to all references.

> What about have the arguments as part of the reference behind `:`. Like this:
> 
> ```
> $ avocado list --resolver-run-executables examples/nrunner/resolvers/exec_runnables_recipe_kind.sh:tap
> ```

I believe this stretches a lot the format of references (even though they are free form).  But we could end up with references such as:

```
my-script.sh:--first-argument --second-argument
```

Which would require quoting, escaping, etc.

Also, your proposal reminds me of a very old and long attempt/discussion to be able to support command line arguments in references to `exec-test` (then called `SIMPLE` tests).  It turned out not being a great idea, because of complexity and corner cases that could not be solved (is something an argument? another executable? etc...).